M2/Macaulay2/m2/typicalvalues.m2

```diff
@@ -15,21 +15,10 @@ typicalValues#(symbol symbol) = Symbol
 typicalValues#(symbol ?) = Boolean
 
 dummy := x -> error("dummy method function called")
-installMethod(symbol #?, HashTable, Thing, Boolean => x -> (dummy x;))
-installMethod(symbol #?, Set, Thing, Boolean => x -> (dummy x;))
-installMethod(symbol #?, Database, String, Boolean => x -> (dummy x;))
+-- TODO: where are these defined in the interpreter?
```

Comment:
I think these `BasicList` methods are the same as the corresponding `List` ones:

* `BasicList #? ZZ`: `subvalueQ` in hashtables.dd
* `BasicList # ZZ`: `subvalue` in hashtables.dd
* `# BasicList`: `lengthFun` in actors.d

Reply:
So is `List` in the interpreter the same as `BasicList` in the top-level?

Reply:
Kind of but not exactly, since `Sequence` also inherits from `BasicList` at top-level, but `List` and `Sequence` are distinct in the interpreter.  (A `Sequence` is just a sequence of `Expr` objects, and a `List` contains a `Sequence` and also keeps track of its class, its hash code, and whether it's mutable.)

So if a compiled function gets an instance of some subclass of `BasicList` as input, there are two possible branches it might end up in inside the `when` clause -- `Sequence` if it's a sequence and `List` for everything else.